### PR TITLE
chore(tests): Use dns.google in dns_lookup tests instead of localhost

### PR DESCRIPTION
**Pull request by jszwedko**
localhost doesn't seem to resolve consistently on MacOS causing test failures locally. Certainly there is some risk relying on dns.google but it seems minimal. We can swap it out with another domain if/as needed. Failures can be seen here: https://github.com/vectordotdev/vrl/pull/875

Also updates the VRL test runner to correctly diff objects.

Easier to review with whitespace changes disabled since I changed the indentation when updating the fixtures.

### CODE REVIEW

File: src/stdlib/dns_lookup.rs [hunk 1]
```diff
@@ -648,11 +696,10 @@ mod tests {
     }
 
     #[test]
-    fn test_custom_class_and_type() {
+    fn test_custom_class() {
         let result = execute_dns_lookup(DnsLookupFn {
-            value: expr!("localhost"),
-            class: expr!("*"),
```
Comment (jszwedko) [inline src/stdlib/dns_lookup.rs:654]: I couldn't get it working with `class` set to `*` so I just changed this test to verify the `type`.

**Comment by jszwedko** [comment]
cc/ @esensar in case you have any ideas on a better way to handle this.

**Review by pront — APPROVED** [review]
Thank you 👍

File: src/test/mod.rs [hunk 2]
```diff
@@ -230,7 +230,7 @@ fn process_result(
                 print!("{} (expectation)", Colour::Red.bold().paint("FAILED"));
 
                 if !config.no_diff {
-                    let want = serde_json::to_string_pretty(&test.result.clone()).unwrap();
+                    let want = serde_json::to_string_pretty(&want_value).unwrap();
```
Comment (pront) [inline src/test/mod.rs:233]: ✨

**Comment by esensar** [comment]
> cc/ @esensar in case you have any ideas on a better way to handle this.

I haven't tested on MacOS, but this is okay, I don't have an idea on how to do it differently.

**Pull request merged**